js/getUserMedia.js

```diff
@@ -95,8 +95,8 @@ function getUserMedia(constraints) {
 		if (typeof constraints.video.deviceId === 'string') {
 			newConstraints.videoDeviceId = constraints.video.deviceId;
 		// Also check sourceId (mangled by adapter.js).
-		} else if (typeof constraints.video.sourceId === 'string') {
-			newConstraints.videoDeviceId = constraints.video.sourceId;
+		} else if (constraints.video.optional && constraints.video.optional[0] && typeof constraints.video.optional[0].sourceId === 'string') {
+			newConstraints.videoDeviceId = constraints.video.optional[0].sourceId;
```

Comment:
I think this is too hacky. It will throw if `constraints.video` is `undefined`, and won't work if `constraints.video.mandatory` are given, or if `sourceId` appears in the second or third entry within `optional`.

Also, please split all those conditions in separate lines to make it more readable:

```js
} else if (
	constraints.video &&
	(
		(constraints.video.optional &&
			constraints.video.optional[0] &&
			typeof constraints.video.optional[0].sourceId === 'string') ||
		(constraints.video.mandatory &&
			constraints.video.mandatory[0] &&
			typeof constraints.video.mandatory[0].sourceId === 'string')
	)
) {
```